tests/console/validate_lvm.pm

```diff
@@ -39,9 +40,39 @@ sub run {
         assert_equals($vol_status, 'ACTIVE', "Volume is Inactive");
     }
 
+    # Sum up the Physical Extents across all Physical Volumes within each Volume group
+    # of a given system. So it gives us : $PEs_in_VGs{$vg} = [@extents_in_vg]
+    # Then do the same for the free extents, so we can compare total PEs with total logical extents.
     record_info('equal extents', 'Verify sum of logical extents corresponds to physical extent size');
-    my $pvTotalPE = script_output q[pvdisplay|grep "Total PE" | awk '{print $3}'];
-    my $pvFreePE  = script_output q[pvdisplay|grep "Free PE" | awk '{print $3}'];
+    my @PV = split(/\n/, script_output qq[pvdisplay |grep "PV Name" |awk '{print \$3}']);
+    my @VG = split(/\n/, script_output qq[vgdisplay |grep "VG Name" |awk '{print \$3}']);
+    my %PEs_in_PVs;
+    my %FEs_in_PVs;
+
+    foreach my $vg (@VG) {
+        my @extents_in_vg;
+        my @free_extents_in_vg;
+        foreach my $pv (@PV) {
+            my $vg_for_pv = script_output qq[pvdisplay $pv |grep "VG Name" |awk '{print \$3}'];
+            chomp($vg_for_pv);
+            if ("$vg_for_pv" eq "$vg") {
+                my $extents_in_pv      = script_output qq[pvdisplay $pv |grep "Total PE" |awk '{print \$3}'];
+                my $free_extents_in_pv = script_output qq[pvdisplay $pv |grep "Free PE" |awk '{print \$3}'];
+                chomp($extents_in_pv, $free_extents_in_pv);
+                push(@extents_in_vg,      $extents_in_pv);
+                push(@free_extents_in_vg, $free_extents_in_pv);
+            }
+            $PEs_in_PVs{$vg} = [@extents_in_vg];
+            $FEs_in_PVs{$vg} = [@free_extents_in_vg];
+        }
+    }
+
+    my $total_PEs;
+    my $total_free_PEs;
+    foreach my $vg (keys %PEs_in_PVs) {
```

Comment:
I'd recommend to not iterate again and perform addition inside previous loop by just accumulating the values in two variables.